File: apps/android/app/src/main/java/com/litter/android/ui/terminal/TerminalControlSequences.kt
package com.litter.android.ui.terminal

internal object TerminalControlSequences {
    const val ESCAPE = "\u001B"
    const val TAB = "\t"
    const val ENTER = "\r"
    const val BACKSPACE = "\u007F"
    const val CTRL_C = "\u0003"
    const val CTRL_D = "\u0004"
    const val CTRL_Z = "\u001A"
    const val ARROW_LEFT = "\u001B[D"
    const val ARROW_UP = "\u001B[A"
    const val ARROW_DOWN = "\u001B[B"
    const val ARROW_RIGHT = "\u001B[C"
    const val HOME = "\u001B[H"
    const val END = "\u001B[F"
    const val PAGE_UP = "\u001B[5~"
    const val PAGE_DOWN = "\u001B[6~"
    const val DROID_MISSIONS_COMMAND = "/missions\n"

    val terminalAccessoryLabels = listOf(
        "Missions",
        "Esc",
        "Tab",
        "Enter",
        "⌫",
        "Home",
        "End",
        "PgUp",
        "PgDn",
        "Ctrl-C",
        "Ctrl-D",
        "Ctrl-Z",
        "←",
        "↑",
        "↓",
        "→",
        "Paste",
        "Interrupt",
        "Close",
    )
}

File: apps/ios/Sources/Litter/Models/TerminalControlSequences.swift
import Foundation

enum TerminalControlSequences {
    static let escape = "\u{1B}"
    static let tab = "\t"
    static let enter = "\r"
    static let backspace = "\u{7F}"
    static let ctrlC = "\u{03}"
    static let ctrlD = "\u{04}"
    static let ctrlZ = "\u{1A}"
    static let arrowLeft = "\u{1B}[D"
    static let arrowUp = "\u{1B}[A"
    static let arrowDown = "\u{1B}[B"
    static let arrowRight = "\u{1B}[C"
    static let home = "\u{1B}[H"
    static let end = "\u{1B}[F"
    static let pageUp = "\u{1B}[5~"
    static let pageDown = "\u{1B}[6~"
    static let droidMissionsCommand = "/missions\n"

    static let terminalAccessoryLabels = [
        "Missions",
        "Esc",
        "Tab",
        "Enter",
        "⌫",
        "Home",
        "End",
        "PgUp",
        "PgDn",
        "Ctrl-C",
        "Ctrl-D",
        "Ctrl-Z",
        "←",
        "↑",
        "↓",
        "→",
        "Paste",
        "Interrupt",
        "Close"
    ]
}
